Refactor Naming Series

shreyashah115 opened the pull request:
- Updated all DocTypes having `autoname=naming_series:` with new Naming Series as per mentioned in #14085.
- Created Property Setter for existing documents having old naming series.

<img width="580" alt="screen shot 2018-08-01 at 5 59 42 pm" src="https://user-images.githubusercontent.com/17617465/43521767-dc4f4f5a-95b4-11e8-9adb-37b1d2d59ed2.png">
<img width="575" alt="screen shot 2018-08-01 at 5 59 58 pm" src="https://user-images.githubusercontent.com/17617465/43521769-dc853d54-95b4-11e8-91ab-d83297fedb90.png">
<img width="577" alt="screen shot 2018-08-01 at 6 00 14 pm" src="https://user-images.githubusercontent.com/17617465/43521770-dcb948ec-95b4-11e8-99c9-091264777204.png">
<img width="576" alt="screen shot 2018-08-01 at 6 00 32 pm" src="https://user-images.githubusercontent.com/17617465/43521772-dce8db34-95b4-11e8-8b2d-052ff9f623d6.png">

```
doctype_series_map = {
	'Additional Salary': 'HR-ADS-.YY.-.MM.-',
	'Appraisal': 'HR-APR-.YY.-.MM.-',
	'Asset': 'ACC-ASS-.YYYY.-',
	'Attendance': 'HR-ATT-.YYYY.-',
	'Auto Repeat': 'SYS-ARP-.YYYY.-',
	'Blanket Order': 'MFG-BLR-.YYYY.-',
	'C-Form': 'ACC-CF-.YYYY.-',
	'Campaign': 'SAL-CAM-.YYYY.-',
	'Clinical Procedure': 'HLC-CPR-.YYYY.-',
	'Course Schedule': 'EDU-CSH-.YYYY.-',
	'Customer': 'CUST-.YYYY.-',
	'Delivery Note': 'MAT-DN-.YYYY.-',
	'Delivery Trip': 'MAT-DT-.YYYY.-',
	'Driver': 'HR-DRI-.YYYY.-',
	'Employee': 'HR-EMP-',
	'Employee Advance': 'HR-EAD-.YYYY.-',
	'Expense Claim': 'HR-EXP-.YYYY.-',
	'Fee Schedule': 'EDU-FSH-.YYYY.-',
	'Fee Structure': 'EDU-FST-.YYYY.-',
	'Fees': 'EDU-FEE-.YYYY.-',
	'Inpatient Record': 'HLC-INP-.YYYY.-',
	'Installation Note': 'MAT-INS-.YYYY.-',
	'Instructor': 'EDU-INS-.YYYY.-',
	'Issue': 'ISS-.YYYY.-',
	'Journal Entry': 'ACC-JV-.YYYY.-',
	'Lab Test': 'HLC-LT-.YYYY.-',
	'Landed Cost Voucher': 'MAT-LCV-.YYYY.-',
	'Lead': 'CRM-LEAD-.YYYY.-',
	'Leave Allocation': 'HR-LAL-.YYYY.-',
	'Leave Application': 'HR-LAP-.YYYY.-',
	'Maintenance Schedule': 'MAT-MSH-.YYYY.-',
	'Maintenance Visit': 'MAT-MVS-.YYYY.-',
	'Material Request': 'MAT-MR-.YYYY.-',
	'Member': 'NPO-MEM-.YYYY.-',
	'Opportunity': 'CRM-OPP-.YYYY.-',
	'Packing Slip': 'MAT-PAC-.YYYY.-',
	'Patient': 'HLC-PAT-.YYYY.-',
	'Patient Encounter': 'HLC-ENC-.YYYY.-',
	'Patient Medical Record': 'HLC-PMR-.YYYY.-',
	'Payment Entry': 'ACC-PAY-.YYYY.-',
	'Payment Request': 'ACC-PRQ-.YYYY.-',
	'Production Plan': 'MFG-PP-.YYYY.-',
	'Project Update': 'PROJ-UPD-.YYYY.-',
	'Purchase Invoice': 'ACC-PINV-.YYYY.-',
	'Purchase Order': 'PUR-ORD-.YYYY.-',
	'Purchase Receipt': 'MAT-PRE-.YYYY.-',
	'Quality Inspection': 'MAT-QA-.YYYY.-',
	'Quotation': 'SAL-QTN-.YYYY.-',
	'Request for Quotation': 'PUR-RFQ-.YYYY.-',
	'Sales Invoice': 'ACC-SINV-.YYYY.-',
	'Sales Order': 'SAL-ORD-.YYYY.-',
	'Sample Collection': 'HLC-SC-.YYYY.-',
	'Shareholder': 'ACC-SH-.YYYY.-',
	'Stock Entry': 'MAT-STE-.YYYY.-',
	'Stock Reconciliation': 'MAT-RECO-.YYYY.-',
	'Student': 'EDU-STU-.YYYY.-',
	'Student Applicant': 'EDU-APP-.YYYY.-',
	'Supplier': 'SUP-.YYYY.-',
	'Supplier Quotation': 'PUR-SQTN-.YYYY.-',
	'Supplier Scorecard Period': 'PU-SSP-.YYYY.-',
	'Timesheet': 'TS-.YYYY.-',
	'Vehicle Log': 'HR-VLOG-.YYYY.-',
	'Warranty Claim': 'SER-WRN-.YYYY.-',
	'Work Order': 'MFG-WO-.YYYY.-'
}
```
```
doctype_series_map = {
	'Activity Cost': 'PROJ-ACC-.#####',
	'Agriculture Task': 'AG-TASK-.#####',
	'Assessment Plan': 'EDU-ASP-.YYYY.-.#####',
	'Assessment Result': 'EDU-RES-.YYYY.-.#####',
	'Asset Movement': 'ACC-ASM-.YYYY.-.#####',
	'Attendance Request': 'HR-ARQ-.YY.-.MM.-.#####',
	'Authorization Rule': 'HR-ARU-.#####',
	'Bank Guarantee': 'ACC-BG-.YYYY.-.#####',
	'Bin': 'MAT-BIN-.YYYY.-.#####',
	'Certification Application': 'NPO-CAPP-.YYYY.-.#####',
	'Certified Consultant': 'NPO-CONS-.YYYY.-.#####',
	'Chat Room': 'CHAT-ROOM-.#####',
	'Compensatory Leave Request': 'HR-CMP-.YY.-.MM.-.#####',
	'Custom Script': 'SYS-SCR-.#####',
	'Employee Benefit Application': 'HR-BEN-APP-.YY.-.MM.-.#####',
	'Employee Benefit Application Detail': '',
	'Employee Benefit Claim': 'HR-BEN-CLM-.YY.-.MM.-.#####',
	'Employee Incentive': 'HR-EINV-.YY.-.MM.-.#####',
	'Employee Onboarding': 'HR-EMP-ONB-.YYYY.-.#####',
	'Employee Onboarding Template': 'HR-EMP-ONT-.#####',
	'Employee Promotion': 'HR-EMP-PRO-.YYYY.-.#####',
	'Employee Separation': 'HR-EMP-SEP-.YYYY.-.#####',
	'Employee Separation Template': 'HR-EMP-STP-.#####',
	'Employee Tax Exemption Declaration': 'HR-TAX-DEC-.YYYY.-.#####',
	'Employee Tax Exemption Proof Submission': 'HR-TAX-PRF-.YYYY.-.#####',
	'Employee Transfer': 'HR-EMP-TRN-.YYYY.-.#####',
	'Event': 'EVENT-.YYYY.-.#####',
	'Exchange Rate Revaluation': 'ACC-ERR-.YYYY.-.#####',
	'GL Entry': 'ACC-GLE-.YYYY.-.#####',
	'Guardian': 'EDU-GRD-.YYYY.-.#####',
	'Hotel Room Reservation': 'HTL-RES-.YYYY.-.#####',
	'Item Price': '',
	'Job Applicant': 'HR-APP-.YYYY.-.#####',
	'Job Offer': 'HR-OFF-.YYYY.-.#####',
	'Leave Encashment': 'HR-ENC-.YYYY.-.#####',
	'Leave Period': 'HR-LPR-.YYYY.-.#####',
	'Leave Policy': 'HR-LPOL-.YYYY.-.#####',
	'Loan': 'ACC-LOAN-.YYYY.-.#####',
	'Loan Application': 'ACC-LOAP-.YYYY.-.#####',
	'Loyalty Point Entry': '',
	'Membership': 'NPO-MSH-.YYYY.-.#####',
	'Packing Slip': 'MAT-PAC-.YYYY.-.#####',
	'Patient Appointment': 'HLC-APP-.YYYY.-.#####',
	'Payment Terms Template Detail': '',
	'Payroll Entry': 'HR-PRUN-.YYYY.-.#####',
	'Period Closing Voucher': 'ACC-PCV-.YYYY.-.#####',
	'Plant Analysis': 'AG-PLA-.YYYY.-.#####',
	'POS Closing Voucher': 'POS-CLO-.YYYY.-.#####',
	'Prepared Report': 'SYS-PREP-.YYYY.-.#####',
	'Program Enrollment': 'EDU-ENR-.YYYY.-.#####',
	'Quotation Item': '',
	'Restaurant Reservation': 'RES-RES-.YYYY.-.#####',
	'Retention Bonus': 'HR-RTB-.YYYY.-.#####',
	'Room': 'HTL-ROOM-.YYYY.-.#####',
	'Salary Structure Assignment': 'HR-SSA-.YY.-.MM.-.#####',
	'Sales Taxes and Charges': '',
	'Share Transfer': 'ACC-SHT-.YYYY.-.#####',
	'Shift Assignment': 'HR-SHA-.YY.-.MM.-.#####',
	'Shift Request': 'HR-SHR-.YY.-.MM.-.#####',
	'SMS Log': 'SYS-SMS-.#####',
	'Soil Analysis': 'AG-ANA-.YY.-.MM.-.#####',
	'Soil Texture': 'AG-TEX-.YYYY.-.#####',
	'Stock Ledger Entry': 'MAT-SLE-.YYYY.-.#####',
	'Student Leave Application': 'EDU-SLA-.YYYY.-.#####',
	'Student Log': 'EDU-SLOG-.YYYY.-.#####',
	'Subscription': 'ACC-SUB-.YYYY.-.#####',
	'Task': 'TASK-.YYYY.-.#####',
	'Tax Rule': 'ACC-TAX-RULE-.YYYY.-.#####',
	'Training Feedback': 'HR-TRF-.YYYY.-.#####',
	'Training Result': 'HR-TRR-.YYYY.-.#####',
	'Travel Request': 'HR-TRQ-.YYYY.-.#####',
	'UOM Conversion Factor': 'MAT-UOM-CNV-.#####',
	'Water Analysis': 'HR-WAT-.YYYY.-.#####',
	'Workflow Action': 'SYS-WACT-.#####',
}
```

shreyashah115 committed:
Patch to create Property Setter for existing naming series

shreyashah115 committed:
Update existing naming series options with new

shreyashah115 committed:
Add patch link to patches.txt

shreyashah115 committed:
Update patch

shreyashah115 committed:
Fix codacy

nabinhait commented on erpnext/patches/v11_0/refactor_naming_series.py:
```diff
@@ -0,0 +1,121 @@
+	return series_to_set
+
+def get_series_to_preserve(doctype):
+	series_to_preserve = frappe.db.sql_list("""select distinct naming_series from `tab{doctype}`""".format(doctype=doctype))
```
exclude blank

nabinhait commented on erpnext/patches/v11_0/refactor_naming_series.py:
```diff
@@ -0,0 +1,121 @@
+	return series_to_preserve
+
+def get_default_series(doctype):
+	default_series = frappe.db.sql("""select naming_series from `tab{doctype}`
```
get default based on meta (custom)

nabinhait commented on erpnext/patches/v11_0/refactor_naming_series.py:
```diff
@@ -0,0 +1,121 @@
+			continue
+		if not series_to_preserve:
+			continue
+		existing_series = frappe.db.get_value('DocField', {'parent': doctype, 'fieldname': 'naming_series'}, 'options').split("\n")
```
get existing series based on meta

nabinhait commented on erpnext/patches/v11_0/refactor_naming_series.py:
```diff
@@ -0,0 +1,121 @@
+
+		series_to_preserve = get_series_to_preserve(doctype)
+		default_series = get_default_series(doctype)
+		if not default_series:
```
not required

nabinhait commented on erpnext/patches/v11_0/refactor_naming_series.py:
```diff
@@ -0,0 +1,121 @@
+
+def set_series(doctype, options, default):
+	make_property_setter(doctype, "naming_series", "options", options, "Text")
+	make_property_setter(doctype, "naming_series", "default", default, "Text")
```
if default

shreyashah115 committed:
Fix as per suggestions

shreyashah115 committed:
Remove old patch for property setter

rmehta commented:
```
doctype_series_map = {
	'Additional Salary': 'HR-ADS-.YY.-.MM.-',
	'Appraisal': 'HR-APR-.YY.-.MM.',
	'Asset': 'ACC-ASS-.YYYY.-',
	'Asset Maintenance Log': 'ACC-AML-.YYYY.-',
	'Asset Repair': 'ACC-ASR-.YYYY.-',
	'Attendance': 'HR-ATT-.YYYY.-',
	'Auto Repeat': 'SYS-ARP-.YYYY.-',
	'Blanket Order': 'MFG-BLR-.YYYY.-',
	'C-Form': 'ACC-CF-.YYYY.-',
	'Campaign': 'SAL-CAM-.YYYY.-',
	'Clinical Procedure': 'HLC-CPR-.YYYY.-',
	'Course Schedule': 'EDU-CSH-.YYYY.-',
	'Customer': 'CUST-.YYYY.-',
	'Delivery Note': 'MAT-DN-.YYYY.-',
	'Delivery Trip': 'MAT-DT-.YYYY.-',
	'Driver': 'HR-DRI-.YYYY.-',
	'Employee': 'HR-EMP-,
	'Employee Advance': 'HR-EAD-.YYYY.-',
	'Expense Claim': 'HR-EXP-.YYYY.-',
	'Fee Schedule': 'EDU-FSH-.YYYY.-',
	'Fee Structure': 'EDU-FST-.YYYY.-',
	'Fees': 'EDU-FEE-.YYYY.-',
	'Inpatient Record': 'HLC-INP-.YYYY.-',
	'Installation Note': 'MAT-INS-.YYYY.-',
	'Instructor': 'EDU-INS-.YYYY.-',
	'Issue': 'ISS-.YYYY.-',
	'Journal Entry': 'ACC-JV-.YYYY.-',
	'Lab Test': 'HLC-LT-.YYYY.-',
	'Landed Cost Voucher': 'MAT-LCV-.YYYY.-',
	'Lead': 'CRM-LEAD-.YYYY.-',
	'Leave Allocation': 'HR-LAL-.YYYY.-',
	'Leave Application': 'HR-LAP-.YYYY.-',
	'Maintenance Schedule': 'MAT-MSH-.YYYY.-',
	'Maintenance Visit': 'MAT-MVS-.YYYY.-',
	'Material Request': 'MAT-MR-.YYYY.-',
	'Member': 'NPO-MEM-.YYYY.-',
	'Opportunity': 'CRM-OPP-.YYYY.-',
	'Packing Slip': 'MAT-PAC-.YYYY.-',
	'Patient': 'HLC-PAT-.YYYY.-',
	'Patient Encounter': 'HLC-ENC-.YYYY.-',
	'Patient Medical Record': 'HLC-PMR-.YYYY.-',
	'Payment Entry': 'ACC-PAY-.YYYY.-',
	'Payment Request': 'ACC-PRQ-.YYYY.-',
	'Production Plan': 'MFG-PP-.YYYY.-',
	'Project Update': 'PROJ-UPD-.YYYY.-',
	'Purchase Invoice': 'ACC-PINV-.YYYY.-',
	'Purchase Order': 'PUR-ORD-.YYYY.-',
	'Purchase Receipt': 'MAT-PRE-.YYYY.-',
	'Quality Inspection': 'MAT-QA-.YYYY.-',
	'Quotation': 'SAL-QTN-.YYYY.-',
	'Request for Quotation': 'PUR-RFQ-.YYYY.-',
	'Sales Invoice': 'ACC-SINV-.YYYY.-',
	'Sales Order': 'SAL-ORD-.YYYY.-',
	'Sample Collection': 'HLC-SC-.YYYY.-',
	'Shareholder': 'ACC-SH-.YYYY.-',
	'Stock Entry': 'MAT-STE-.YYYY.-',
	'Stock Reconciliation': 'MAT-RECO-.YYYY.-',
	'Student': 'EDU-STU-.YYYY.-',
	'Student Applicant': 'EDU-APP-.YYYY.-',
	'Supplier': 'SUP-.YYYY.-',
	'Supplier Quotation': 'PUR-SQTN-.YYYY.-',
	'Supplier Scorecard Period': 'PU-SSP-.YYYY.-',
	'Timesheet': 'TS-.YYYY.-',
	'Vehicle Log': 'HR-VLOG-.YYYY.-',
	'Warranty Claim': 'SER-WRN-.YYYY.-',
	'Work Order': 'MFG-WO-.YYYY.-'
}
```

```
Activity Cost - PROJ-ACC
Agriculture Task - AG-TASK-
Assessment Plan - EDU-ASP-
Assessment Result - EDU-RES-
Asset Movement - ACC-ASM-
Attendance Request - HR-ARQ-
Authorization Rule - HR-ARU
Bank Guarantee - ACC-BG-
Bin - MAT-BIN-
Certification Application - NPO-CAPP-
Certified Consultant - NPO-CONS-
Chat Room - CHAT-ROOM-
Compensatory Leave Request - HR-CMP-
Custom Script - SYS-SCR-
Employee Benefit Application - HR-BEN-APP-
Employee Benefit Application Detail (hash)
Employee Benefit Claim - HR-BEN-CLM-
Employee Incentive - HR-EINV-
Employee Onboarding - HR-EMP-ONB-
Employee Onboarding Template - HR-EMP-ONT-
Employee Promotion - HR-EMP-PRO-
Employee Separation - HR-EMP-SEP
Employee Separation Template - HR-EMP-STP
Employee Tax Exemption Declaration - HR-TAX-DEC-
Employee Tax Exemption Proof Submission HR-TAX-PRF
Employee Transfer - HR-EMP-TRN
Event - EVENT-
Exchange Rate Revaluation - ACC-ERR
GL Entry - ACC-GLE-
Guardian - EDU-GRD-
Hotel Room Reservation - HOT-RES
Item Price - (hash)
Job Applicant - HR-APP-
Job Offer - HR-OFF
Leave Encashment - HR-ENC-
Leave Period - HR-LPR-
Leave Policy - HR-LPOL-
Loan - ACC-LOAN-
Loan Application - ACC-LOAP
Loyalty Point Entry - (hash)
Membership - NPO-MSH-
Packing Slip - MAT-PAC-
Patch Log (don't change)
Patient Appointment - HLC-APP
Payment Terms Template Detail   (hash)
Payroll Entry - HR-PRUN-
Period Closing Voucher - ACC-PCV-
Plant Analysis - AG-PLA-
POS Closing Voucher - POS-CLO-
Prepared Report - SYS-PREP-
Program Enrollment - EDU-ENR-
Quotation Item (hash)
Restaurant Reservation - HOT-RES
Retention Bonus - HR-RTB-
Room - HOT-ROOM
Salary Structure Assignment - HR-SSA-
Sales Taxes and Charges (hash)
Share Transfer - ACC-SHT-
Shift Assignment - HR-SHA-
Shift Request - HR-SHR
SMS Log - SYS-SMS-
Soil Analysis - AG-ANA-
Soil Texture - AG-TEX-
Stock Ledger Entry - MAT-SLE-
Student Leave Application - EDU-SLA-
Student Log - EDU-SLOG
Subscription - ACC-SUB
Task - TASK-
Tax Rule - HR-TAX-RULE-
Training Feedback - HR-TRF-
Training Result - HR-TRR-
Travel Request - HR-TRQ-
UOM Conversion Factor - 
Water Analysis - HR-WAT-
Workflow Action - SYS-WACT-
```

shreyashah115 committed:
Improve as per suggested

- Modified the naming
- Also refactored naming for doctypes having autoname with series
- Added a patch to make property setter for autoname

nabinhait commented:
@shreyashah115 The travis is breaking

shreyashah115 committed:
Fix Travis

shreyashah115 committed:
Rebase with develop

shreyashah115 commented:
@nabinhait Have fixed Travis for patch related failures.

nabinhait committed:
Merge branch 'develop' into refactor-series
